pyproject.toml

```diff
@@ -0,0 +1,49 @@
+[tool.poetry]
+name = "cookiecutter-python-template"
+version = "0.1.0"
+description = """Cookiecutter template for a python projects."""
+authors = ["Parth Barhanpurkar <parth.barhanpurkar@analog.com>"]
+readme = "README.md"
+
+[tool.poetry.dependencies]
+python = ">=3.10,<3.12"
```

Comment:
To start with, we can keep one version. I have set it to 3.11. But I think this could lead to version issues when users dont have the right version on their system. I am not completely sure

Reply:
You should either install pyenv so that you can run multiple Python distros at the same time, or use a project devcontainer that has the correct Python version installed in it

Reply:
Got it. I am keeping the python version ">=3.11 and < 3.13". The version of python in github actions is set to 3.12.